### PR TITLE
Allow submission of checkin information via json

### PR DESCRIPTION
**Pull request by puiterwijk**
This means we still allow pickle submissions as a stopgap until we
manage to convince admins to move to json submissions.

Related: CVE-2016-1000003
Signed-off-by: Patrick Uiterwijk <puiterwijk@redhat.com>

### CODE REVIEW

File: mirrormanager2/xmlrpc.py [hunk 1]
```diff
@@ -41,7 +42,11 @@
 
 @XMLRPC.register
 def checkin(pickledata):
-    config = pickle.loads(bz2.decompress(base64.urlsafe_b64decode(pickledata)))
+    uncompressed = bz2.decompress(base64.urlsafe_b64decode(pickledata))
+    try:
+        config = json.loads(uncompressed)
+    except ValueError:
+        config = pickle.loads(uncompressed)
```
Comment (bowlofeggs) [inline mirrormanager2/xmlrpc.py]: Should we file a reminder issue somewhere so we don't forget to remove the pickle support soon?

Comment (puiterwijk) [inline mirrormanager2/xmlrpc.py]: I have a reminder in my calendar, but sounds good.

**Comment by bowlofeggs** [comment]
This looks good to me!

+1

**Comment by puiterwijk** [comment]
Thanks. #174 filed.

File: mirrormanager2/xmlrpc.py (hunk 1 above)
Comment (pypingou) [inline mirrormanager2/xmlrpc.py]: I had an idea, could we log here that a pickle got submitted/read so maybe we can monitor over the coming days how many pickle we are processing and see if that goes up or down?

**Pull request merged**